airbyte-integrations/connectors/source-tiktok-marketing/source_tiktok_marketing/components/advertiser_ids_partition_router.py

```diff
@@ -0,0 +1,70 @@
+# Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+
+import json
+from typing import Any, Iterable, Mapping
+
+import dpath.util
+from airbyte_cdk.sources.declarative.partition_routers.substream_partition_router import SubstreamPartitionRouter
+from airbyte_cdk.sources.declarative.types import StreamSlice
+
+
+class MultipleAdvertiserIdsPartitionRouter(SubstreamPartitionRouter):
```

Comment:
Let's rename this to `MultipleAdvertiserIdsPerPartition` and the below one `SingleAdvertiserIdPerPartition`. I think we want to really make it clear that what is happening is that it is how many advertiser_ids are in one partition. Because when named `SingleAdvertiserIdPartitionRouter` makes it sound like we only ever get one advertiser id.

Reply:
Thanks, renaming makes sense to me.

Regarding MultipleAdvertiserIdsPerPartition, I forgot to mention that `advertiser_ids` is required request param and request to /advertiser/info/ can't be performed with `advertiser_id` as a string. So approach with MultipleAdvertiserIdsPerPartition is only way to read advertisers stream properly.